packages/protocol/contracts/signal/SignalService.sol

```diff
@@ -148,23 +160,35 @@ contract SignalService is EssentialContract, ISignalService {
         HopProof[] memory hopProofs = abi.decode(_proof, (HopProof[]));
         if (hopProofs.length == 0) revert SS_EMPTY_PROOF();
 
+        uint256 lenLessOne;
+        unchecked {
+            lenLessOne = hopProofs.length - 1;
+        }
+        uint64[] memory trace = new uint64[](lenLessOne);
+
```

Comment:
Here (`verifySignalReceived`) i think there is no need to check these things as this is a helper view function to the relayer and not used for proving.

Reply:
good point

Reply:
I think verifySignalReceived should behavice the same so if it returns true, then proveSignalReceived should succeed.